### DISCUSSION TITLE
How to detect when a feature was added WinUI3?

### DISCUSSION THREAD

**Discussion by EugenM88** [Mapsui/Mapsui#2361] [General]
Greetings, first of all thank you very much for the awesome community that created and supported this wonderful mapping system.
I have been trying to implement this system to only select 1 single point, however apparently there is no limit to how many points you can insert, that is fine as long as i can somehow find an event whenever a feature is added to a layer, and exactly here is my struggle.

I cannot find any example or anything related to my issue, i need a trigger event for whenever a new feature is added and/or removed from the said map.

Here is my test code for reference

```
 MyMap.Map.Layers.Add(OpenStreetMap.CreateTileLayer());
 MyMap.Map.Layers.Add(new WritableLayer() { Name = "Location Layer", Style = new VectorStyle() { Fill = new Mapsui.Styles.Brush(new Mapsui.Styles.Color(0, 255, 255)), Line = new Pen(new Mapsui.Styles.Color(0, 0, 255), 3), Outline = new Pen(new Mapsui.Styles.Color(50, 50, 50), 2) } });
 EditManager mapsUIEditManager = new EditManager();
 mapsUIEditManager.EditMode = EditMode.AddPoint;
 mapsUIEditManager.Layer = (MyMap.Map.Layers[1] as WritableLayer);
 
EditManipulation editManipulation = new EditManipulation();
mapsUIEditManager.Layer.DataChanged += Layer_DataChanged;//not triggering when user adds a point
 if (mapsUIEditManager.Layer.Extent != null)
 {
     MRect extent = mapsUIEditManager.Layer.Extent!.Grow(mapsUIEditManager.Layer.Extent.Width * 0.2);
     MyMap.Map.Navigator.ZoomToBox(extent)    
 }
 
 MyMap.Map.Widgets.Add(new EditingWidget(MyMap, mapsUIEditManager, editManipulation));
```
**EDIT: Found an  issue with my code, now my question has been updated.** 
The only way i could find to detect if something has changed is like this, however im not sure this is the designed approach.
Since after reading most of the MapsUI website sample documentation i still have the same question... 
could anyone please give me some info or links to any documentation regarding the best way to implement this scenario?

Thank you in advance for all of your help!

**Comment by pauldendulk** [comment]
The editing functionality is a first version and in the experimental phase it is better to take things under your own control.

You could use a custom widget. Clone the source code of Mapsui and look for CustomWidgetSample.cs and CustomWidgetSkiaRenderer.cs (you need to register the renderer). If you have that working look at the EditingWidget. Copy that code to your custom widget and modify it to your own needs.
